service/src/main/java/org/cbioportal/service/impl/TreatmentServiceImpl.java

```diff
@@ -25,7 +25,7 @@ private Pair<List<String>, List<String>> filterIds(List<String> sampleIds, List<
         }
         Set<String> studiesWithTreatments = studyIds.stream()
             .distinct()
-            .filter(studyId -> treatmentRepository.studyIdHasTreatments(studyId, key))
+            .filter(studyId -> treatmentRepository.hasTreatmentData(Collections.singletonList(studyId), key))
```

Comment:
This might not be appropriate for this PR but do you think this could be done as a SQL query?

Reply:
Could you write a ticket for this. Even with a new database we may still need to address this issue.

Reply:
Created; https://github.com/cBioPortal/cbioportal/issues/10269